### PR TITLE
refactor(docs): clean up and restructure

### PR DESCRIPTION
**Pull request by khorne3**
Edits, minor refactoring of code-server docs.

### CODE REVIEW

File: docs/CONTRIBUTING.md
Comment (jsjoeio) [inline docs/CONTRIBUTING.md:14]: nit: If "Integration Tests" is title-cased, shouldn't "End-to-end tests" be too? i.e. "End-to-end Tests"

File: docs/CONTRIBUTING.md
Comment (jsjoeio) [inline docs/CONTRIBUTING.md:16]: nit: I'm worried changing this from "Modifications" to "Modifying" changes the meaning. For additional context, this section is meant for maintainers/contributors to document any modifications they've made to VS Code instead of instructions for modifying VS Code.

File: docs/CONTRIBUTING.md
Comment (jsjoeio) [inline docs/CONTRIBUTING.md:27]: ? Will this break the link being on two lines? Any reason is shouldn't be on one line? (too long?)

File: docs/CONTRIBUTING.md
Comment (jsjoeio) [inline docs/CONTRIBUTING.md:75]: ^See comment above

File: docs/CONTRIBUTING.md
Comment (jsjoeio) [inline docs/CONTRIBUTING.md:136]: Ah I see you're only capitalizing the first word. I think either works as long as we stay consistent

File: docs/CONTRIBUTING.md
Comment (jsjoeio) [inline docs/CONTRIBUTING.md:186]: ^See above

**Review by jsjoeio — CHANGES_REQUESTED** [review]
This is fantastic! Thanks for doing this work @khorne3 

Mostly questions/nits for consistency.

Other request: do you mind squashing your commits into one and using a semantic commit? e.g. "refactor(docs): clean up and restructure

File: docs/CONTRIBUTING.md
Comment (khorne3) [reply docs/CONTRIBUTING.md:16]: Thanks for the clarification -- I'll change this back

File: docs/CONTRIBUTING.md
Comment (khorne3) [reply docs/CONTRIBUTING.md:27]: I think the link should be fine, but the reason this sometimes happens is Product Docs enforces a strict line length so my IDE automatically does stuff like this across all MD files

**Review by jsjoeio — APPROVED** [review]
You rock! Thanks again for doing this! I'll approve, feel free to merge when ready!

Han solo saying yahoo

**Pull request merged**
